Sources/LanguageServerProtocol/SupportTypes/WorkspaceSettings.swift

```diff
@@ -59,6 +56,11 @@ public struct ClangWorkspaceSettings: Codable, Hashable {
     self.compilationDatabasePath = compilationDatabasePath
     self.compilationDatabaseChanges = compilationDatabaseChanges
   }
+
+  internal func isValid() -> Bool {
+    return (self.compilationDatabasePath != nil && self.compilationDatabaseChanges == nil)
+            || (self.compilationDatabasePath == nil && self.compilationDatabaseChanges != nil)
```

Comment:
How do you feel about making this a `switch` so we don’t need to repeat the property names?

```suggestion
    switch (compilationDatabasePath, compilationDatabaseChanges) {
      case (nil, .some), (.some, nil): return true
      default: return false
    }
```